pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
 	}
 }
 
-// CreateSCIMIntegration implements schema.CreateFunc.
-func CreateSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// CreateContextSCIMIntegration implements schema.CreateFunc.
+func CreateContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
 	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	enabled := d.Get("enabled").(bool)
+	scimClient := d.Get("scim_client").(string)
+	runAsRole := d.Get("run_as_role").(string)
 
-	stmt := snowflake.NewSCIMIntegrationBuilder(name).Create()
-
-	// Set required fields
-	stmt.SetRaw(`TYPE=SCIM`)
-	stmt.SetString(`SCIM_CLIENT`, d.Get("scim_client").(string))
-	stmt.SetString(`RUN_AS_ROLE`, d.Get("provisioner_role").(string))
+	req := sdk.NewCreateScimSecurityIntegrationRequest(id, enabled, sdk.ScimSecurityIntegrationScimClientOption(scimClient), sdk.ScimSecurityIntegrationRunAsRoleOption(runAsRole))
```

Comment:
Instead of sdk.ScimSecurityIntegrationScimClientOption(scimClient) We should have a `ToScimClient` method on the SDK level (now casing may be different, and it may cause some hard-to-spot errors in the future)

Reply:
Done, also for `runAsRole`.